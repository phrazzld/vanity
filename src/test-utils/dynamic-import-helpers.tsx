/**
 * Test helpers for Next.js dynamic imports
 *
 * These utilities help test components that use Next.js dynamic()
 * for code splitting and lazy loading.
 */

import React from 'react';
import { waitFor } from '@testing-library/react';
import preloadAll from 'jest-next-dynamic';

/**
 * Preload all dynamic imports before running tests
 * Call this in beforeAll() hooks for test suites that use dynamic imports
 */
export async function preloadDynamicImports(): Promise<void> {
  await preloadAll();
}

/**
 * Wait for a dynamic component to load and render
 * Useful when testing components that show loading states
 */
export async function waitForDynamicComponent(
  getElement: () => HTMLElement | null,
  options?: { timeout?: number }
): Promise<void> {
  await waitFor(
    () => {
      const element = getElement();
      expect(element).toBeInTheDocument();
    },
    { timeout: options?.timeout || 5000 }
  );
}

/**
 * Mock for next/dynamic that immediately renders the component
 * without async loading behavior - useful for unit tests
 */
export function createSyncDynamicMock() {
  return jest.fn().mockImplementation((importFunc: () => Promise<any>, _options?: any) => {
    // If there's a loading component, we'll ignore it for sync tests
    // Return a component that renders immediately
    const Component = jest.fn().mockImplementation((props: any) => {
      const MockComponent = require(
        importFunc.toString().match(/import\(['"](.+)['"]\)/)?.[1] || ''
      );
      return MockComponent.default ? MockComponent.default(props) : MockComponent(props);
    });

    return Component;
  });
}

/**
 * Mock for next/dynamic that simulates async loading
 * Provides better control over loading states in tests
 */
export function createAsyncDynamicMock() {
  return jest.fn().mockImplementation((importFunc: () => Promise<any>, options?: any) => {
    const LoadingComponent = options?.loading || (() => null);

    return function MockDynamicComponent(props: any) {
      const [isLoading, setIsLoading] = React.useState(true);
      const [Component, setComponent] = React.useState<any>(null);

      React.useEffect(() => {
        // Simulate async import
        void importFunc().then(module => {
          setComponent(() => module.default || module);
          setIsLoading(false);
        });
      }, []);

      if (isLoading) {
        return <LoadingComponent />;
      }

      return Component ? <Component {...props} /> : null;
    };
  });
}
