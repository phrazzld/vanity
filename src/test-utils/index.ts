// src/test-utils/index.ts
import React, { useState } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeContext } from '@/app/context/ThemeContext';
import userEvent from '@testing-library/user-event';

// ===========================================================================
// Theme Provider Test Setup
// ===========================================================================

/**
 * A custom theme provider for tests that doesn't use browser APIs
 */
export function TestThemeProvider({
  children,
  initialTheme = 'light',
}: {
  children: React.ReactNode;
  initialTheme?: 'light' | 'dark';
}) {
  const [isDarkMode, setIsDarkMode] = useState(initialTheme === 'dark');

  const toggleDarkMode = () => setIsDarkMode(prev => !prev);

  return React.createElement(
    ThemeContext.Provider,
    { value: { isDarkMode, toggleDarkMode } },
    React.createElement(
      'div',
      { 
        'data-testid': 'theme-provider',
        'data-theme': isDarkMode ? 'dark' : 'light'
      },
      children
    )
  );
}

/**
 * Custom render method that includes the TestThemeProvider wrapper
 * Simplified from https://testing-library.com/docs/react-testing-library/setup
 */
type CustomRenderOptions = Omit<RenderOptions, 'wrapper'> & {
  themeMode?: 'light' | 'dark';
  // Add other custom render options here as needed
};

export function renderWithTheme(
  ui: React.ReactElement,
  options?: CustomRenderOptions
) {
  const { themeMode = 'light', ...renderOptions } = options || {};
  
  return render(ui, {
    wrapper: ({ children }) => (
      React.createElement(TestThemeProvider, { initialTheme: themeMode }, children)
    ),
    ...renderOptions,
  });
}

// Setup user-event with standard configuration
export function setupUser(options = { delay: null }) {
  return userEvent.setup(options);
}

// ===========================================================================
// Mock Response Helpers
// ===========================================================================

// Mock Fetch Response Helper
export function createMockResponse<T>(data: T, status = 200, statusText = 'OK') {
  return {
    ok: status >= 200 && status < 300,
    status,
    statusText,
    json: jest.fn().mockResolvedValue(data),
    text: jest.fn().mockResolvedValue(JSON.stringify(data)),
  };
}

// Create a failed response
export function createMockErrorResponse(
  status = 500,
  statusText = 'Server Error',
  message = 'An error occurred'
) {
  return {
    ok: false,
    status,
    statusText,
    json: jest.fn().mockResolvedValue({ error: message }),
    text: jest.fn().mockResolvedValue(JSON.stringify({ error: message })),
  };
}

// Mocks the fetch for testing API calls
export function mockFetch<T>(data: T, status = 200) {
  const mockResponse = createMockResponse(data, status);
  const fetchMock = jest.fn().mockResolvedValue(mockResponse);
  // Cast window to unknown first to avoid TypeScript errors
  (window as unknown as { fetch: typeof fetchMock }).fetch = fetchMock;
  return { mockResponse, mockFetch: fetchMock };
}

// ===========================================================================
// Test Utilities
// ===========================================================================

// Helper to wait for a condition
export function waitForCondition(
  callback: () => boolean | Promise<boolean>,
  { timeout = 1000, interval = 50 } = {}
): Promise<void> {
  return new Promise((resolve, reject) => {
    const startTime = Date.now();

    const checkCondition = async () => {
      const result = await callback();
      if (result) {
        resolve();
        return;
      }

      const elapsedTime = Date.now() - startTime;
      if (elapsedTime > timeout) {
        reject(new Error(`Timed out waiting for condition after ${timeout}ms`));
        return;
      }

      setTimeout(checkCondition, interval);
    };

    checkCondition();
  });
}

// Hook testing with theme context
import { renderHook, RenderHookOptions, RenderHookResult } from '@testing-library/react';

/**
 * Custom renderHook method that includes the ThemeProvider wrapper
 */
export function renderHookWithTheme<Result, Props>(
  hook: (props: Props) => Result,
  options?: Omit<RenderHookOptions<Props>, 'wrapper'> & {
    themeMode?: 'light' | 'dark';
  }
): RenderHookResult<Result, Props> {
  const { themeMode = 'light', ...renderOptions } = options || {};
  
  return renderHook(hook, {
    wrapper: ({ children }) => (
      React.createElement(TestThemeProvider, { initialTheme: themeMode }, children)
    ),
    ...renderOptions,
  });
}

// Export everything from testing library for convenience
export * from '@testing-library/react';
export { userEvent };
