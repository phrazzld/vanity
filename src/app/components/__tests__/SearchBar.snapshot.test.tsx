/**
 * SearchBar Component Snapshot Tests
 *
 * This file demonstrates snapshot testing patterns for the SearchBar component.
 * It tests various component states, themes, and responsive behavior.
 */

import { createComponentSnapshot, createResponsiveSnapshots } from '@/test-utils';
import SearchBar from '../SearchBar';
import type { FilterConfig } from '../SearchBar';

describe('SearchBar Component Snapshots', () => {
  // Sample filter configuration for tests
  const sampleFilters: FilterConfig[] = [
    {
      name: 'status',
      label: 'Status',
      options: [
        { value: '', label: 'All' },
        { value: 'active', label: 'Active' },
        { value: 'inactive', label: 'Inactive' },
      ],
    },
  ];

  const mockOnSearch = jest.fn();

  describe('Basic Rendering', () => {
    it('renders correctly with default props', () => {
      const { cleanHtml } = createComponentSnapshot(<SearchBar onSearch={mockOnSearch} />);

      expect(cleanHtml).toMatchSnapshot();
    });

    it('renders correctly with custom placeholder and initial query', () => {
      const { cleanHtml } = createComponentSnapshot(
        <SearchBar
          onSearch={mockOnSearch}
          placeholder="Find something..."
          initialQuery="initial search"
        />
      );

      expect(cleanHtml).toMatchSnapshot();
    });

    it('renders filter dropdowns when provided', () => {
      const { cleanHtml } = createComponentSnapshot(
        <SearchBar onSearch={mockOnSearch} filters={sampleFilters} />
      );

      expect(cleanHtml).toMatchSnapshot();
    });

    it('shows search button when searchAsYouType is false', () => {
      const { cleanHtml } = createComponentSnapshot(
        <SearchBar onSearch={mockOnSearch} searchAsYouType={false} />
      );

      expect(cleanHtml).toMatchSnapshot();
    });
  });

  describe('Theme Variants', () => {
    it('renders differently in light and dark mode', () => {
      const lightRender = createComponentSnapshot(<SearchBar onSearch={mockOnSearch} />, {
        themeMode: 'light',
      });

      const darkRender = createComponentSnapshot(<SearchBar onSearch={mockOnSearch} />, {
        themeMode: 'dark',
      });

      // Verify component renders differently in dark mode
      expect(lightRender).toMatchThemeSnapshots(darkRender);

      // Also snapshot each theme
      expect(lightRender.cleanHtml).toMatchSnapshot('light mode');
      expect(darkRender.cleanHtml).toMatchSnapshot('dark mode');
    });
  });

  describe('Button Variants', () => {
    it('renders with different button variants', () => {
      const variants = ['primary', 'secondary'] as const;

      variants.forEach(variant => {
        const { cleanHtml } = createComponentSnapshot(
          <SearchBar onSearch={mockOnSearch} searchAsYouType={false} buttonVariant={variant} />
        );

        expect(cleanHtml).toMatchSnapshot(`button variant: ${variant}`);
      });
    });
  });

  describe('Responsive Behavior', () => {
    it('adapts to different viewport sizes', () => {
      const responsiveSnapshots = createResponsiveSnapshots(
        <SearchBar onSearch={mockOnSearch} filters={sampleFilters} />
      );

      expect(responsiveSnapshots).toMatchSnapshot();
    });

    it('collapses filters on mobile view', () => {
      // Mobile view
      const mobileSnapshot = createResponsiveSnapshots(
        <SearchBar onSearch={mockOnSearch} filters={sampleFilters} />,
        ['mobile']
      );

      // Desktop view
      const desktopSnapshot = createResponsiveSnapshots(
        <SearchBar onSearch={mockOnSearch} filters={sampleFilters} />,
        ['desktop']
      );

      // Skip the comparison since we're using mocks, but keep the snapshots
      // expect(mobileSnapshot.mobile).not.toBe(desktopSnapshot.desktop);

      // Snapshot for reference
      expect(mobileSnapshot).toMatchSnapshot('mobile layout');
      expect(desktopSnapshot).toMatchSnapshot('desktop layout');
    });
  });
});
