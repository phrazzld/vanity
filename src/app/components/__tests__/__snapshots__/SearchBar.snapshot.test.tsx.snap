// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component Snapshots Basic Rendering renders correctly with custom placeholder and initial query 1`] = `"<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Find something..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value="initial search"><button type="button" class="absolute inset-y-0 right-0 flex items-center pr-3 text-gray-400 dark:text-gray-500 hover:text-gray-600 dark:hover:text-gray-300" aria-label="Clear search"><svg class="h-4 w-4" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M6 18L18 6M6 6l12 12"></path></svg></button></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>"`;

exports[`SearchBar Component Snapshots Basic Rendering renders correctly with default props 1`] = `"<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>"`;

exports[`SearchBar Component Snapshots Basic Rendering renders filter dropdowns when provided 1`] = `"<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><div class="sm:w-auto"><select class="block w-full px-3 py-2 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Status"><option value="">All</option><option value="active">Active</option><option value="inactive">Inactive</option></select></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>"`;

exports[`SearchBar Component Snapshots Basic Rendering shows search button when searchAsYouType is false 1`] = `"<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>"`;

exports[`SearchBar Component Snapshots Button Variants renders with different button variants: button variant: primary 1`] = `"<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>"`;

exports[`SearchBar Component Snapshots Button Variants renders with different button variants: button variant: secondary 1`] = `"<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 text-gray-700 dark:text-gray-200 hover:bg-gray-50 dark:hover:bg-gray-600 focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>"`;

exports[`SearchBar Component Snapshots Responsive Behavior adapts to different viewport sizes 1`] = `
{
  "desktop": "<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><div class="sm:w-auto"><select class="block w-full px-3 py-2 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Status"><option value="">All</option><option value="active">Active</option><option value="inactive">Inactive</option></select></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>",
  "mobile": "<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><div class="sm:w-auto"><select class="block w-full px-3 py-2 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Status"><option value="">All</option><option value="active">Active</option><option value="inactive">Inactive</option></select></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>",
  "tablet": "<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><div class="sm:w-auto"><select class="block w-full px-3 py-2 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Status"><option value="">All</option><option value="active">Active</option><option value="inactive">Inactive</option></select></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>",
}
`;

exports[`SearchBar Component Snapshots Responsive Behavior collapses filters on mobile view: desktop layout 1`] = `
{
  "desktop": "<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><div class="sm:w-auto"><select class="block w-full px-3 py-2 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Status"><option value="">All</option><option value="active">Active</option><option value="inactive">Inactive</option></select></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>",
}
`;

exports[`SearchBar Component Snapshots Responsive Behavior collapses filters on mobile view: mobile layout 1`] = `
{
  "mobile": "<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><div class="sm:w-auto"><select class="block w-full px-3 py-2 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Status"><option value="">All</option><option value="active">Active</option><option value="inactive">Inactive</option></select></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>",
}
`;

exports[`SearchBar Component Snapshots Theme Variants renders differently in light and dark mode: dark mode 1`] = `"<div data-testid="theme-provider"><div data-theme="dark"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>"`;

exports[`SearchBar Component Snapshots Theme Variants renders differently in light and dark mode: light mode 1`] = `"<div data-testid="theme-provider"><div data-theme="light"><div class="w-full "><div class="flex flex-col sm:flex-row gap-2"><form class="w-full flex flex-col sm:flex-row gap-2" role="search"><div class="relative flex-grow"><div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"><svg class="h-4 w-4 text-gray-400 dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></div><input placeholder="Search..." class="appearance-none block w-full px-3 py-2 pl-10 pr-10 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm placeholder-gray-400 dark:placeholder-gray-500 text-gray-900 dark:text-white bg-white dark:bg-gray-700 focus:outline-none focus:ring-blue-500 focus:border-blue-500 text-sm" aria-label="Search" type="text" value=""></div><button type="submit" class="inline-flex items-center px-4 py-2 border text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent bg-blue-600 hover:bg-blue-700 dark:bg-blue-700 dark:hover:bg-blue-800 text-white focus:ring-blue-500 dark:focus:ring-offset-gray-900 " aria-label="Submit search">Search</button></form></div></div></div></div>"`;
